src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -222,10 +243,19 @@ private void BuildGlobbedLinkTags(IDictionary<string, string> attributes, TagHel
         private void BuildFallbackBlock(TagHelperContent builder)
         {
             EnsureGlobbingUrlBuilder();
-            var fallbackHrefs = GlobbingUrlBuilder.BuildUrlList(FallbackHref, FallbackHrefInclude, FallbackHrefExclude);
+            var fallbackHrefs =
+                GlobbingUrlBuilder.BuildUrlList(FallbackHref, FallbackHrefInclude, FallbackHrefExclude).ToArray();
 
             if (fallbackHrefs.Any())
             {
+                if (FileVersion == "true")
+                {
+                    for (var i=0; i < fallbackHrefs.Count(); i++)
+                    {
+                        fallbackHrefs[i] += "?v=" + GetFileContentEncodedString(fallbackHrefs[i]);
```

Comment:
We shouldn't be doing manual URL building logic here if possible. See if the types in `Microsoft.AspNet.Http` help with this logic, e.g. `PathString`, `QueryString`, `FragmentString`. Look at `Microsoft.AspNet.Http.Extensions.UriHelper` to help with splitting the URL up into its typed parts. It doesn't look like we have the functionality in there to add an additional query value and then emit the full URL again though, so you may have to add that to `Microsoft.AspNet.WebUtilities.QueryHelpers` or similar. Check with @loudej